### PR TITLE
fix: storybook's defaultValue warning

### PR DESCRIPTION
**Pull request by muhajirdev**
TSIA

### CODE REVIEW

**Comment by muhajirdev** [comment]
image

no more warning about the deprecation

**Comment by muhajirdev** [comment]
image
no more warning about the deprecation

File: .storybook/main.js [hunk 1]
```diff
@@ -19,7 +19,15 @@ module.exports = {
...
-    reactDocgen: false
+    check: false,
+    checkOptions: {},
+    // ideally we use `react-docgen-typescript`. But there's still some issue, related to webpack 5.
+    // so we use `react-docgen` for now
+    reactDocgen: 'react-docgen', 
+    reactDocgenTypescriptOptions: {
+      shouldExtractLiteralValuesFromEnum: true,
+      propFilter: (prop) => (prop.parent ? !/node_modules/.test(prop.parent.fileName) : true),
+    },
```
Comment (tifa2UP) [inline .storybook/main.js:30]: Why are we doing this?

**Review by tifa2UP — APPROVED** [review]

File: .storybook/main.js (hunk 1 above)
Comment (muhajirdev) [reply .storybook/main.js:30]: image

This `react-docgen` infers some information from the source code. Such as `default` value in the table.

**Comment by osamasayed** [comment]
LGTM

**Review by osamasayed — APPROVED** [review]

**Pull request merged**